### DISCUSSION TITLE
.aae files not getting imported

### DISCUSSION THREAD

**Discussion by martinjr85** [photoprism/photoprism#3089] [General]
PLEASE PROCEED ONLY IF YOU ARE ABSOLUTELY SURE THAT THIS IS NOT A TECHNICAL SUPPORT INCIDENT AND/OR POSSIBLY A PROBLEM WITH SOME OTHER SOFTWARE YOU ARE USING. VISIT <https://photoprism.app/kb/getting-support> TO LEARN MORE ABOUT OUR SUPPORT OPTIONS. THANK YOU FOR YOUR CAREFUL CONSIDERATION!

---------------------------------------------------------------------------

#### 1. What is not working as documented?

After exporting some unmodified originals from macOS Photos, I am unable to see the .aae files getting imported into Photoprism as expected

#### 2. How can we reproduce it?

Attempt to import the attached files into Photoprism using the import process. Additionally, check the "Move Files" checkbox.
Observe the .aae files are not imported and moved into the /originals folder.

#### 3. What behavior do you expect?

I expect the files to be imported into Photoprism and moved into the /originals folder.

#### 4. What could be the cause of your problem?

I have no idea. It must be a bug within the software

#### 5. Can you provide us with example files for testing, error logs, or screenshots?

Sample files included

[Archive.zip](https://github.com/photoprism/photoprism/files/10377513/Archive.zip)

#### 6. Which software versions do you use?

(a) PhotoPrism Architecture & Build Number: AMD64, ARM64, ARMv7,...
Build 221118-e58fee0fb ARMv8

(b) Database Type & Version: MariaDB, MySQL, SQLite,...
MariaDB

(c) Operating System Types & Versions: Linux, Windows, Android,...
Raspbian OS

(d) Browser Types & Versions: Firefox, Chrome, Safari on iPhone,...
Chrome

(e) Ad Blockers, Browser Plugins, and/or Firewall Software?
n/a

You can find the version/build number of the app in *Settings* by scrolling to the bottom. Note that MySQL 8 support has been discontinued, see system requirements at <https://docs.photoprism.app/getting-started/#system-requirements>.

*Always provide database and operating system details if it is a backend, import, or indexing issue. Should it be a frontend issue, at a minimum we require you to provide web browser and operating system details. When reporting a performance problem, we ask that you provide us with complete information about your environment, as there may be more than one cause.*

#### 7. On what kind of device is PhotoPrism installed?

This is especially important if you are reporting a performance, import, or indexing issue. You can skip this if you're reporting a problem you found in our public demo, or if it's a completely unrelated issue, such as incorrect page layout.

(a) Device / Processor Type: Raspberry Pi 4, Intel Core i7-3770, AMD Ryzen 7 3800X,...
Raspberry Pi 4

(b) Physical Memory & Swap Space in GB
8GB RAM, 4096MB swap

(c) Storage Type: HDD, SSD, RAID, USB, Network Storage,...
HDD

(d) Anything else that might be helpful to know?
n/a

*Always provide device, memory, and storage details if you have a backend, performance, import, or indexing issue.*

#### 8. Do you use a Reverse Proxy, Firewall, VPN, or CDN?
Traefik 2.9.4

**Comment by graciousgrey** [comment]
The folder you are trying to import contains only AAE files and no related original image files. 

Sidecar files must be located in the same folder as their related originals and share the same name. Otherwise PhotoPrism does not know that they belong together.

When you import a folder that contains both the sidecar and the related original, e.g.

- photos/IMG_123.aae 
- photos/IMG_123.jpg

both will be moved to the originals folder.

**Reply by martinjr85** [reply]
Thanks for the help @graciousgrey. I have exported the originals from macOS Photos again and have all files: .heic, .xmp, .aae
However, I dont want to upload these here in public so I will send you a private message on Gitter.

**Reply by martinjr85** [reply]
It appears the .xmp and .heic files were imported and only .aae was not imported. I do notice these .aae files are created only for the _portrait_ photos that were taken from the iPhone

**Reply by graciousgrey** [reply]
All files have been in the same folder and have the exact same name (except from the file-extension) ?

**Reply by martinjr85** [reply]
The Photos app actually doesn't export the .aae files with the same name for some reason. Looks like an upper case O is added to the filename.
https://user-images.githubusercontent.com/5271069/211586756-3fa17613-114e-4324-8e07-28ed9734bd6e.png

**Reply by graciousgrey** [reply]
Then that's the issue. If you remove the O it should work as expected.

**Reply by martinjr85** [reply]
Thanks @graciousgrey. What a bummer that Apple is doing this because I have quite a few files that I need to modify. If I decide to change the filenames and attempt a re-upload only the unimported .aae files, I suppose they will find their proper place in the /originals folder during the import process?  I can see filenames are changed during import so just want to make sure they are associated to their respective originals if that makes sense.

**Reply by lastzero** [reply]
No, because the sidecar files must be imported along with the main files. At best, you can make sure that both are present in the import folder, are named correctly, and then import them so that PhotoPrism can recognize the connection with the already imported files.

**Reply by martinjr85** [reply]
Gotcha. In that case I will try exporting from Photos and importing into Photoprism again and make sure that .heic, .xmp, and .aae are all present

**Reply by martinjr85** [reply]
It's kinda annoying that Apple is adding a O to the filename

**Reply by martinjr85** [reply]
I can confirm that after re-importing the .heic, .aae, and .xmp files all having matching filenames that all files are imported and moved out of the import folder.  I will have to remember to take an extra step and remove the O each time I export from macOS Photos and import into Photoprism.

**Reply by martinjr85** [reply]
However, I don't see the .aae file associated with the photos from the UI. Should I expect to see the aae files from the UI?

**Reply by lastzero** [reply]
They might not be shown as they are not media files. As long as they are copied that should be fine. Maybe you see them in Library > Originals.

**Reply by martinjr85** [reply]
I actually dont see them under Library -> Originals in the UI but I do see them in the /originals path on the server :). Thanks for the help, Michael.
